src/Components/Blazor/WebAssembly.Authentication/src/AuthenticationManager.cs

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// A component that handles remote authentication operations in an application.
+    /// </summary>
+    /// <typeparam name="TAuthenticationState">The user state type persisted while the operation is in progress. It must be serializable.</typeparam>
+    public class AuthenticationManager<TAuthenticationState> : ComponentBase where TAuthenticationState : RemoteAuthenticationState
+    {
+        private string _message;
+
+        /// <summary>
+        /// Gets or sets the <see cref="RemoteAuthenticationActions"/> action the component needs to handle.
+        /// </summary>
+        [Parameter] public string Action { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <typeparamref name="TAuthenticationState"/> to be preserved during the authentication operation.
+        /// </summary>
+        [Parameter] public TAuthenticationState AuthenticationState { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="RenderFragment"/> with the UI to display while <see cref="RemoteAuthenticationActions.Login"/> is being handled.
+        /// </summary>
+        [Parameter] public RenderFragment LoginFragment { get; set; } = DefaultLoginFragment;
```

Comment:
When we went through API review prior to 3.0, we removed the suffix "Fragment" from all the `RenderFragment`-type parameters in public APIs. Examples: https://github.com/dotnet/aspnetcore/blob/master/src/Components/Authorization/src/AuthorizeViewCore.cs#L28-L39

Could all the params here follow a similar naming pattern?

* For items that represent an action in progress, prefer the gerund form ("-ing") of the name. Example: `LoggingIn` or `RedirectingToRegistration`
* For items that represent a completed action, prefer the past-tense form of the name. Example: `LoginFailed`
* For anything else, try to pick something that makes sense (without the `Fragment` suffix) in declarative usage

Also, do you think there might be too many things here for people to customize? It's not too bad for people building English-language UIs since they can get away with leaving things uncustomized if they don't know what they mean. But for someone building a non-English UI, they *must* read our docs to figure out what everything here means and how to represent it in their language, which is a big cliff to make them climb.

As a possible simplification, what if every one of the "redirection is in progress" states was represented by a single `Redirecting` fragment with default text `Redirecting...`. Then if someone wants to put in more details about why the redirection is happening, they can do this:

```razor
<Redirecting>
@if (action === "LoggingIn`)
{
    <p>Hold tight - logging you in!</p>
} else if (action === "SomethingElse") {
    <p>... etc ... </p>
}
</Redirecting>
```

Developers who don't want to customize each message (e.g., because they don't know what they all mean or why they would occur) can just stick with the default `Redirecting...` message, perhaps translated into their own language.

Reply:
Likewise all the states representing "processing a return action" could be represented by a single `<ProcessingResponse>` fragment with default text `Processing...`. Again it could by customized in more detail using `@if (action === ` for developers who feel that adds some value to their application.

Reply:
This was changed.